sphinx/util/inventory.py

```diff
@@ -138,6 +138,11 @@ def load_v2(
                 continue
             if location.endswith('$'):
                 location = location[:-1] + name
+            if type == 'std:label':
+                # Normalize names for `:std:label` to match
+                # `XRefRole(lowercase=True)` for `ref` and `numref` roles.
+                # See https://github.com/sphinx-doc/sphinx/issues/12008
+                name = name.lower()
```

Comment:
Although I think it's unlikely to occur, I think that we should add a check to see whether an existing entry exists in `invdata` after performing this lowercasing, and emit an `info`-level message if so.

It _shouldn't_ really happen, but if it does it'd be nice to have some way to track that down and report any problems in the affect `.inv` file(s).

Reply:
Sure! Actually, that's not even *that* unlikely to happen: If there are two headings in the project that only differ by case and whatever documentation-generator is being used doesn't disambiguate them, you'd end up with a conflict. I actually might have written documentations in Documenter that would have had that problem. Good catch!

> emit an info-level message if so.

I would even consider that worthy of a warning-level message. Can you remind me how to do that? I haven't done much Python in a while, and forgot the details of a lot of Sphinx' internals: do we have something specific in Sphinx to emit messages at various levels?

Reply:
> I would even consider that worthy of a warning-level message. Can you remind me how to do that? I haven't done much Python in a while, and forgot the details of a lot of Sphinx' internals: do we have something specific in Sphinx to emit messages at various levels?

A warning-level message was my initial thought too, but from experience: Sphinx is used by a _lot_ of downstream consumers, and a number of them use quite strict warning-as-error settings.  I don't think we'd want to break any projects based on what _other_ projects (and potentially other documentation engines) are serving from their `objects.inv` file, so I changed my mind and would recommend `info`-level.  I probably should've explained all that before suggesting it!

The usual pattern for logging in this codebase is:

```py
from sphinx.util import logging
...
logger = logging.getLogger(__name__)
...
logger.info(__('info-level log message'))
```

The `__(...)` method implies translatable strings (localization) - we have some documentation about how that works here: https://www.sphinx-doc.org/en/master/internals/contributing.html#translations

(currently not every log message is internationalized.. but most of them are, and we should probably localize the remainders)